src/actions/exchangeActions.js

```diff
@@ -77,14 +80,45 @@ export const takeOfferAction = (
 ) => {
   return async (dispatch: Dispatch, getState: GetState) => {
     connectExchangeService(getState());
+    const {
+      accounts: { data: accounts },
+      exchange: { exchangeSupportedAssets },
+    } = getState();
+
+    const fromAsset = exchangeSupportedAssets.find(a => a.symbol === fromAssetCode);
+    const toAsset = exchangeSupportedAssets.find(a => a.symbol === toAssetCode);
+
+    if (!fromAsset || !toAsset) {
+      Toast.show({
+        title: 'Exchange service failed',
+        type: 'warning',
+        message: 'Could not find asset',
+      });
+      callback({});
+      return;
+    }
+
+    let { address: fromAssetAddress } = fromAsset;
+    const { decimals: fromAssetDecimals } = fromAsset;
+    let { address: toAssetAddress } = toAsset;
+
+    // we need PROD assets' addresses in order to get offers when on ropsten network
+    // as v2 requests require to provide addresses not tickers
+    if (__DEV__) {
```

Comment:
Yeah, as I've wrote in the description, exchange works on QA (since v2 is only deployed there) with mainnet (as we do not have ropsten assets offers).

The problem is that even if one sets mainnet (homestead) as an active network for environment, as a user that is registered on QA, he/she will only get ropsten assets and we need to provide mainnet assets addresses for exchange requests - this is one of breaking changes that was needed to be handled. 
Previously we needed just assets tickers (symbols). That's why we could easily get offers on QA previously.

Reply:
Then why isn't the `exchangeSupportedAssets` always production assets?

Reply:
Well, it's a tricky part - those are not actually supported on QA wallet.
Exchange is not fully working here as well. On QA we can only get offers, not proceed with those to actually exchange assets.